### DISCUSSION TITLE
Security of Scoop packages

### DISCUSSION THREAD

**Discussion by avoutsas67** [ScoopInstaller/Scoop#5383] [Help and Questions]
Dear all, 

I started using Scoop very recently and I find it extremely useful. 

Does anyone have experience in using it in a corporate (security controlled) environment, where the security and provenance of the packages, at least from the ScoopInstaller buckets , would need to be demonstrated?

I would very much appreciate if you could explain if there is a process in place to ensure the provenance and the security of the packages from the ScoopInstaller bucket or if there are alternative buckets that offer this.  

Many thanks in advance for your response, 

Achilleas

**Comment by r15ch13** [comment]
The maintainers review packages that are added.
Updating these packages is automated with [ScoopInstaller/GithubActions](https://github.com/ScoopInstaller/GithubActions)
Except for programs hosted in [ScoopInstaller/Binary](https://github.com/ScoopInstaller/Binary) everything gets downloaded from the original source.

Overall, it is probably equal to manually downloading and installing a program.

If a program is compromised, only the stored hash in the manifest would cause the installation to fail.
However, if a new version is created with malicious code and a matching hash (modified web page, hash file recreated), then the automatic update process would accept it and update the manifest.

**Comment by rashil2000** [comment]
The packages added to Main bucket are vetted by the maintainers - fully open source, or very popular command-line applications are added there. Individual binaries and downloaded files are not verified, because they come from the first-party source itself, other sources are not allowed. 
The Extras bucket also follows a similar procedure, but the criteria are more relaxed there. 

That said, the best way to use Scoop in internal organizations is to create a private bucket, and only allow installing applications from that bucket. That bucket can only contain programs that your organization allows/audits. It is very easy to create your own bucket, see here: https://github.com/ScoopInstaller/Scoop/wiki/Buckets#creating-your-own-bucket. You can copy manifests from the Main bucket (or create your own manifests) and then make the bucket private. To download from private buckets, users will need to set a GitHub auth token using the [`scoop config`](https://github.com/ScoopInstaller/Scoop/blob/master/libexec/scoop-config.ps1#L83-L86) command.

**Reply by bt4ibwem8** [reply]
Can everybody add the new package to main or Extras bucket or only the maintainers? For example if i miss some few usefull programs which are not included in these buckets can i add them (for example to Extras bucket)?

**Reply by rashil2000** [reply]
If they meet the criteria for that bucket (mentioned in the package-request issue template), then yes, anybody can make a PR for adding these programs to the bucket.
